bcda/api/requests_test.go

```diff
@@ -229,6 +231,107 @@ func (s *RequestsTestSuite) TestRunoutDisabled() {
 }
 
+func (s *RequestsTestSuite) TestDataTypeAuthorization() {
+	acoA, _ := regexp.Compile(`A\d{4}`)
+	acoB, _ := regexp.Compile(`B\d{4}`)
+	acoC, _ := regexp.Compile(`C\d{4}`)
+	acoD, _ := regexp.Compile(`D\d{4}`)
+
+	dataTypeMap := map[string]DataType{
+		"Coverage":             {Adjudicated: true},
+		"Patient":              {Adjudicated: true},
+		"ExplanationOfBenefit": {Adjudicated: true},
+		"Claim":                {Adjudicated: false, PreAdjudicated: true},
+		"ClaimResponse":        {Adjudicated: false, PreAdjudicated: true},
+	}
+
+	h := NewHandler(dataTypeMap, "/v2/fhir/", "v2")
+
+	// Use a mock to ensure that this test does not generate artifacts in the queue for other tests
+	enqueuer := &queueing.MockEnqueuer{}
+	enqueuer.On("AddJob", mock.Anything, mock.Anything).Return(nil)
+	h.Enq = enqueuer
+	h.supportedDataTypes = dataTypeMap
+
+	h.Svc = MockService{
+		acoConfig: map[*regexp.Regexp]*service.ACOConfig{
+			acoA: {
+				Model:              "Model A",
+				Pattern:            "A\\d{4}",
+				PerfYearTransition: "01/01",
+				LookbackYears:      10,
+				Disabled:           false,
+				Data:               []string{"adjudicated", "pre-adjudicated"},
+			},
+			acoB: {
+				Model:              "Model B",
+				Pattern:            "B\\d{4}",
+				PerfYearTransition: "01/01",
+				LookbackYears:      10,
+				Disabled:           false,
+				Data:               []string{"adjudicated"},
+			},
+			acoC: {
+				Model:              "Model C",
+				Pattern:            "C\\d{4}",
+				PerfYearTransition: "01/01",
+				LookbackYears:      10,
+				Disabled:           false,
+				Data:               []string{"pre-adjudicated"},
+			},
+			acoD: {
+				Model:              "Model D",
+				Pattern:            "D\\d{4}",
+				PerfYearTransition: "01/01",
+				LookbackYears:      10,
+				Disabled:           false,
+				Data:               []string{},
+			},
+		},
+	}
+
+	client.SetLogger(log.API) // Set logger so we don't get errors later
+
+	jsonBytes, _ := json.Marshal("{}")
+
+	tests := []struct {
+		name         string
+		cmsId        string
+		resources    []string
+		expectedCode int
+	}{
+		{"Auth Adj/Pre-Adj, Request Adj/Pre-Adj", "A0000", []string{"Claim", "Patient"}, http.StatusAccepted},
+		{"Auth Adj, Request Adj", "B0000", []string{"Patient"}, http.StatusAccepted},
+		{"Auth Adj, Request Pre-Adj", "B0000", []string{"Claim"}, http.StatusBadRequest},
+		{"Auth Pre-Adj, Request Adj", "C0000", []string{"Patient"}, http.StatusBadRequest},
+		{"Auth Pre-Adj, Request Pre-Adj", "C0000", []string{"Claim"}, http.StatusAccepted},
+		{"Auth None, Request Adj", "D0000", []string{"Patient"}, http.StatusBadRequest},
+		{"Auth None, Request Pre-Adj", "D0000", []string{"Claim"}, http.StatusBadRequest},
+	}
+
+	for _, test := range tests {
+		s.T().Run(test.name, func(t *testing.T) {
+			w := httptest.NewRecorder()
+			r, _ := http.NewRequest("GET", "http://bcda.ms.gov/api/v2/Group/$export", bytes.NewReader(jsonBytes))
```

Comment:
If we are sending an empty body, we can leave this as nil. But not a big deal to change.